configuration/src/test/java/io/airlift/configuration/TestConfigurationFactory.java

```diff
@@ -440,15 +441,15 @@ public void testUsedPropertiesWithFailure()
         configurationFactory.registerConfigurationClasses(ImmutableList.of(binder -> configBinder(binder).bindConfig(AnnotatedSetter.class)));
         List<Message> messages = configurationFactory.validateRegisteredConfigurationProvider();
         assertMessagesMatch(messages, ImmutableList.of(".*Invalid value 'invalid' for type boolean \\(property 'boolean-value'\\).*AnnotatedSetter.*"));
-        assertThat(configurationFactory.getUsedProperties()).isEqualTo(properties.keySet());
+        assertThat(configurationFactory.getUsedProperties().stream().map(ConfigPropertyMetadata::name).collect(toImmutableSet())).isEqualTo(properties.keySet());
     }
 
     private static Injector createInjector(Map<String, String> properties, Module module, WarningsMonitor warningsMonitor)
     {
         ConfigurationFactory configurationFactory = new ConfigurationFactory(properties, warningsMonitor);
         configurationFactory.registerConfigurationClasses(ImmutableList.of(module));
         List<Message> messages = configurationFactory.validateRegisteredConfigurationProvider();
-        assertThat(configurationFactory.getUsedProperties()).hasSameElementsAs(properties.keySet());
+        assertThat(configurationFactory.getUsedProperties().stream().map(ConfigPropertyMetadata::name)).hasSameElementsAs(properties.keySet());
```

Comment:
you can do: 

`assertThat(configurationFactory.getUsedProperties())
   .extracting(ConfigPropertyMetadata::name)
   .hasSameElementsAs()`

wrap this for improved readability

Reply:
here and elsewhere